File: apps/web/src/types/meeting.ts
export interface Meeting {
  id: string;
  project_id: string;
  number: number;
  formatted_number: string;
  title: string;
  meeting_type: string;
  status: string;
  scheduled_date?: string | null;
  start_time?: string | null;
  end_time?: string | null;
  location?: string | null;
  virtual_provider?: string | null;
  virtual_link?: string | null;
  attendees: string[];
  agenda?: string | null;
  minutes?: string | null;
  action_items: Record<string, unknown>[];
  recurring: boolean;
  recurrence_rule?: string | null;
  recurrence_end_date?: string | null;
  parent_meeting_id?: string | null;
  minutes_published: boolean;
  minutes_published_at?: string | null;
  created_by?: string | null;
  created_by_name?: string | null;
  created_at: string;
  updated_at: string;
}

export interface MeetingCreateInput {
  title: string;
  meeting_type?: string;
  scheduled_date?: string;
  start_time?: string;
  end_time?: string;
  location?: string;
  virtual_provider?: string;
  virtual_link?: string;
  attendees?: string[];
  agenda?: string;
  recurring?: boolean;
  recurrence_rule?: string;
}

export interface MeetingUpdateInput {
  title?: string;
  meeting_type?: string;
  scheduled_date?: string;
  start_time?: string;
  end_time?: string;
  location?: string;
  attendees?: string[];
  agenda?: string;
  minutes?: string;
  action_items?: Record<string, unknown>[];
}

File: apps/api/app/schemas/meeting.py
from datetime import date, datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.schemas.common import PaginationMeta


class ActionItemEntry(BaseModel):
    description: str
    assigned_to: Optional[UUID] = None
    due_date: Optional[date] = None


class MeetingCreate(BaseModel):
    title: str
    meeting_type: str = "PROGRESS"
    scheduled_date: Optional[date] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    virtual_provider: Optional[str] = None
    virtual_link: Optional[str] = None
    attendees: list[UUID] = []
    agenda: Optional[str] = None
    recurring: bool = False
    recurrence_rule: Optional[str] = None
    recurrence_end_date: Optional[datetime] = None


class MeetingUpdate(BaseModel):
    title: Optional[str] = None
    meeting_type: Optional[str] = None
    scheduled_date: Optional[date] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    virtual_provider: Optional[str] = None
    virtual_link: Optional[str] = None
    attendees: Optional[list[UUID]] = None
    agenda: Optional[str] = None
    minutes: Optional[str] = None
    action_items: Optional[list[ActionItemEntry]] = None
    recurring: Optional[bool] = None
    recurrence_rule: Optional[str] = None
    recurrence_end_date: Optional[datetime] = None


class PublishMinutesRequest(BaseModel):
    create_todos: bool = False


class MeetingResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    project_id: UUID
    number: int
    formatted_number: str
    title: str
    meeting_type: str
    status: str
    scheduled_date: Optional[date] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    location: Optional[str] = None
    virtual_provider: Optional[str] = None
    virtual_link: Optional[str] = None
    attendees: list = []
    agenda: Optional[str] = None
    minutes: Optional[str] = None
    action_items: list = []
    recurring: bool = False
    recurrence_rule: Optional[str] = None
    recurrence_end_date: Optional[datetime] = None
    parent_meeting_id: Optional[UUID] = None
    minutes_published: bool = False
    minutes_published_at: Optional[datetime] = None
    created_by: Optional[UUID] = None
    created_by_name: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class MeetingListResponse(BaseModel):
    data: list[MeetingResponse]
    meta: PaginationMeta
